### PR TITLE
LUI-170 - Not redirecting to requested page after redirected to login…

### PR DESCRIPTION
**Pull request by mseaton**
… page

This adds a new request filter that sets the redirect url on the session to the currently requested url if the user is not authenticated and it is not already set.

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/web/filter/RedirectAfterLoginFilter.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	/**
+	 * @see Filter#doFilter(ServletRequest, ServletResponse, FilterChain)
+	 */
+	public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain) throws IOException, ServletException {
+		try {
+			HttpServletRequest httpReq = (HttpServletRequest) req;
+			String requestURI = httpReq.getRequestURI();
+			if (!Context.isAuthenticated()) {
+				if ("GET".equalsIgnoreCase(httpReq.getMethod()) && !requestURI.contains("login.")) {
+					if (req.getAttribute(OPENMRS_LOGIN_REDIRECT_HTTPSESSION_ATTR) == null) {
```
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/web/filter/RedirectAfterLoginFilter.java:48]: Does this check actually work? I can see we use it elsewhere, but it seems like this should be something like:

```java
req.getSession(false).getAttribute(OPENMRS_LOGIN_REDIRECT_HTTPSESSION_ATTR) == null
```

When I try naïvely to call:

```java
req.getSession().setAttribute("MyAttribute", "SomeValue");
```

And then do:

```java
req.getAttribute("MyAttribute")
```

The response is still `null`.

**Review by ibacher — COMMENTED** [review]
This mostly looks good to me and fixes some rather annoying behaviour.

File: omod/src/main/java/org/openmrs/module/web/filter/RedirectAfterLoginFilter.java (hunk 1 above)
Comment (mseaton) [reply omod/src/main/java/org/openmrs/module/web/filter/RedirectAfterLoginFilter.java]: Oops, sorry, yeah, good catch.  I was initially setting a request attribute and then changed it when I recognized it was supposed to be a session attribute, but didn't change this.

File: omod/src/main/java/org/openmrs/module/web/filter/RedirectAfterLoginFilter.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+						String queryParams = httpReq.getQueryString();
+						String redirectUrl = requestURI + (StringUtils.isNotBlank(queryParams) ? "?" + queryParams : "");
+						httpReq.getSession().setAttribute(OPENMRS_LOGIN_REDIRECT_HTTPSESSION_ATTR, redirectUrl);
+						if (log.isDebugEnabled()) {
+							log.debug("Set " + OPENMRS_LOGIN_REDIRECT_HTTPSESSION_ATTR + " = " + redirectUrl);
+						}
```
[the remark this answers is not in the document]
Comment (mseaton) [reply omod/src/main/java/org/openmrs/module/web/filter/RedirectAfterLoginFilter.java]: Makes sense.

File: omod/src/main/java/org/openmrs/module/web/filter/RedirectAfterLoginFilter.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+					}
+				}
+			}
+		}
+		catch (Exception e) {
+			// Ignore errors here to prevent any failures from disrupting service
```
[the remark this answers is not in the document]
Comment (mseaton) [reply omod/src/main/java/org/openmrs/module/web/filter/RedirectAfterLoginFilter.java]: Wouldn't hurt

**Review by mseaton — COMMENTED** [review]
Thanks @ibacher - great suggestions.  Those are pushed above.

**Pull request merged**